src/python/pants/engine/internals/build_files_test.py

```diff
@@ -802,3 +802,45 @@ def test_invalid_build_file_env_vars(caplog, target_adaptor_rule_runner: RuleRun
             ),
         ],
     )
+
+
+def test_build_file_parse_error(target_adaptor_rule_runner: RuleRunner) -> None:
+    target_adaptor_rule_runner.write_files(
+        {
+            "src/bad/BUILD": dedent(
+                """\
+                mock_tgt(
+                  name="foo"
+                  tags=[]
+                )
+                """
+            ),
+        },
+    )
+    with pytest.raises(ExecutionError, match='File "src/bad/BUILD", line 3'):
```

Comment:
It's not clear why this test passes... maybe it doesn't?

Reply:
it does pass, there's a syntax error on line 3. Changing the expected error to read `line 4` yields a failed test with:
```
E       pants.engine.internals.scheduler.ExecutionError: 1 Exception encountered:
E
E       Engine traceback:
E         in select
E           ..
E         in pants.engine.internals.build_files.find_target_adaptor
E           ..
E
E       Traceback (most recent call last):
E         File "/private/var/folders/y2/6t5bbk554wv5h6w_qlgpcpnm0000gr/T/pants-sandbox-WUOIQs/src/python/pants/engine/internals/selectors.py", line 626, in native_engine_generator_send
E           res = rule.send(arg) if err is None else rule.throw(throw or err)
E         File "/private/var/folders/y2/6t5bbk554wv5h6w_qlgpcpnm0000gr/T/pants-sandbox-WUOIQs/src/python/pants/engine/internals/build_files.py", line 417, in find_target_adaptor
E           address_family = await Get(AddressFamily, AddressFamilyDir(address.spec_path))
E         File "/private/var/folders/y2/6t5bbk554wv5h6w_qlgpcpnm0000gr/T/pants-sandbox-WUOIQs/src/python/pants/engine/internals/selectors.py", line 118, in __await__
E           result = yield self
E         File "/private/var/folders/y2/6t5bbk554wv5h6w_qlgpcpnm0000gr/T/pants-sandbox-WUOIQs/src/python/pants/engine/internals/selectors.py", line 626, in native_engine_generator_send
E           res = rule.send(arg) if err is None else rule.throw(throw or err)
E         File "/private/var/folders/y2/6t5bbk554wv5h6w_qlgpcpnm0000gr/T/pants-sandbox-WUOIQs/src/python/pants/engine/internals/build_files.py", line 321, in parse_address_family
E           all_env_vars = [
E         File "/private/var/folders/y2/6t5bbk554wv5h6w_qlgpcpnm0000gr/T/pants-sandbox-WUOIQs/src/python/pants/engine/internals/build_files.py", line 322, in <listcomp>
E           await _extract_env_vars(
E         File "/private/var/folders/y2/6t5bbk554wv5h6w_qlgpcpnm0000gr/T/pants-sandbox-WUOIQs/src/python/pants/engine/internals/build_files.py", line 243, in _extract_env_vars
E           env_vars = (*BUILDFileEnvVarExtractor.get_env_vars(file_content), *extra_env)
E         File "/private/var/folders/y2/6t5bbk554wv5h6w_qlgpcpnm0000gr/T/pants-sandbox-WUOIQs/src/python/pants/engine/internals/build_files.py", line 207, in get_env_vars
E           obj.visit(ast.parse(file_content.content, file_content.path))
E         File "/Users/andreas.stenius/.pyenv/versions/3.9.13/lib/python3.9/ast.py", line 50, in parse
E           return compile(source, filename, mode, flags,
E         File "src/bad/BUILD", line 3
E           tags=[]
E           ^
E       SyntaxError: invalid syntax

src/python/pants/engine/internals/scheduler.py:497: ExecutionError
```

Reply:
(due to a missing comma on line 2)